accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,320 @@
+- Garbage collect: See below
+- Install workload manifest
+  - This implementation doesn't need to support downloading the .nupkg manifest, as the advertised manifest from the NuGet global packages folder suffices
+  - Calculate workload manifest package ID: `<ManifestID>.<SdkFeatureBand>`
+  - Look for specified version of workload manifest package in global packages folder
+  - Copy contents of `data` folder from package to `<DOTNET ROOT>/sdk-manifests/<SdkFeatureBand>/<ManifestID>/`
+    - To support rollback: first copy existing contents to temporary folder
+    - To rollback: delete folder in SDK and copy contents back from temporary folder
+- Read / write workload installation record
+  - Installation records should be empty files with workload ID name under `sdk-manifests/<SdkFeatureBand>/.installedworkloads/`
+  - Read: List files in band directory
+  - Write: Create file
+  - Delete: Delete file
+
+### Workload Pack installation records and garbage collection
+
+A workload pack installation record will be a folder with the path `<DOTNET ROOT>/sdk-manifests/.installedpacks/v1/<Pack ID>/<Pack Version>/`.  The `v1` element in the path gives us some flexibility to change the format in a later version if we need to.  A reference count for an SDK feature band will be a `<SdkFeatureBand>` folder under the installation record folder.  When there are no SDK feature band reference count folders under a workload pack installation record folder, then the workload pack can be deleted from disk, and then the whole workload pack installation record folder can be deleted.
+
+Thus, the garbage collection process will be as follows:
+
+- Get installed SDK versions and map them to SDK feature bands (ie 6.0.200-preview6 or 6.0.205 would map to 6.0.200)
+- Get installed workloads for current SDK feature band
+- Using current workload manifest, map from currently installed workloads to all pack IDs and versions that should be installed for the current band
+- For each pack / version combination where there is an installation record folder (ie `<DOTNET ROOT>/sdk-manifests/.installedpacks/v1/<Pack ID>/<Pack Version>/`):
+  - Delete any reference count folders that correspond to a band that is not an installed SDK
+  - If the pack ID and version is not in the list that should be installed for the current band, and there is a reference count folder for the current band, then delete that reference count folder
+  - If there are now no reference count folders under the workload pack installation record folder, then the workload pack can be garbage collected
+    - Delete the workload pack file or folder from disk
+    - Delete the workload pack installation record folder: `<DOTNET ROOT>/sdk-manifests/.installedpacks/v1/<Pack ID>/<Pack Version>/`
+    - Also delete the `<Pack ID>` folder if there are no longer any `<Pack Version>` folders under it
+ 
+## Standalone Windows IAL Implementation
+
+- Unit of installation: Workload packs
```

Comment:
This sequence might need to be slightly differently, but perhaps that's an implementation detail based on how the elevation model works. There are two separate processes involved. The client (non-elevated) needs to evaluate all the MSI actions and generate an a plan which based on these would include multiple actions for each MSI. Then, once the server is running (elevated), the client will send requests to only perform operations that require elevation, e.g. installing an MSI, updating a registry key.

We'd also need to think how reboots are handled (especially if a reboot is triggered during install), restart manager sessions when then there are in-use files that lock out installs,

Since there is no UI other than the CLI, we need to consider things like cancellations, which may require some amount of interaction.

Accessibility is also a potential concern for CLI installs.